src/Groups/homomorphisms.jl

```diff
@@ -568,6 +576,57 @@ function isomorphism(::Type{FPGroup}, G::GAPGroup; on_gens::Bool=false)
    end::GAPGroupHomomorphism{typeof(G), FPGroup}
 end
 
+# If `G` is not a full pc group then switch to a full pc group in Oscar.
+function isomorphism(T::Type{PcGroup}, G::GAPGroup; on_gens::Bool=false)
+   # Known isomorphisms are cached in the attribute `:isomorphisms`.
+   isos = get_attribute!(Dict{Tuple{Type, Bool}, Any}, G, :isomorphisms)::Dict{Tuple{Type, Bool}, Any}
+   return get!(isos, (T, false)) do
+     if on_gens
+       # Throw an exception if `gens(G)` is not a polycyclic sequence for `G`.
+       # Create a group in `IsPcGroup` if `G` is finite,
+       # and a group in `IsPcpGroup` otherwise.
+       if is_finite(G)
+         fam = GAP.Globals.ElementsFamily(GAP.Globals.FamilyObj(G.X))::GapObj
+         Ggens = GapObj([x.X for x in gens(G)])::GapObj
+         Cpcgs = GAP.Globals.PcgsByPcSequence(fam, Ggens)::GapObj
+         CC = GAP.Globals.PcGroupWithPcgs(Cpcgs)::GapObj
+         CCpcgs = GAP.Globals.FamilyPcgs(CC)::GapObj
+         f = GAP.Globals.GroupHomomorphismByImages(G, CC, Cpcgs, CCpcgs)::GapObj
+         return GAPGroupHomomorphism(G, T(CC), f)
+       else
+         f = GAP.Globals.IsomorphismPcpGroup(G.X)::GapObj
```

Comment:
```suggestion
         f = GAP.Globals.IsomorphismPcpGroup(GapObj(G))::GapObj
```
possibly also similar (i.e. `FOO.X` to `GapObj(FOO)`) elsewhere?

Reply:
O.k., I change all those `.X` introduced in this pull request; if I change also others then there will likely be conflicts.